lib/Parser/Parse.cpp

```diff
@@ -2247,9 +2247,10 @@ void Parser::ReduceDeferredScriptLength(size_t chars)
 
 void Parser::EnsureStackAvailable()
 {
-    if (!m_scriptContext->GetThreadContext()->IsStackAvailable(Js::Constants::MinStackCompile))
+    bool isInterrupt = false;
+    if (!m_scriptContext->GetThreadContext()->IsStackAvailable(Js::Constants::MinStackCompile, &isInterrupt))
     {
-        Error(ERRnoMemory);
+        Error(isInterrupt ? E_ABORT : VBSERR_OutOfStack);
```

Comment:
I feel like there is one more place you can change as well, ParseArgList function.

Reply:
@akroshg Changed to new error code in commit 2.